runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -77,6 +78,137 @@ void J9::RecognizedCallTransformer::process_java_lang_Class_IsAssignableFrom(TR:
+
+   // Now we have the originalBlock, the fallback block and the merge block (tail block). The original and fallback blocks have commoned nodes.
+   // The fallback block and tail block have commoned nodes as well. The call result will be an astore as well.
+   // If this worked as expected, then the treetop after the call node should be an astore to a temp  slot. So assert that it is, and then grab
+   // a reference to the temp slot so we can store to it ourselves in the forward path.
+   TR::Node *resultStoreNode = treetop->getNextTreeTop()->getNode();
+   TR::SymbolReference *tempSlotForCallResult = NULL;
+   if (resultStoreNode->getOpCode().getOpCodeValue() == TR::astore)
+      {
+      tempSlotForCallResult = resultStoreNode->getSymbolReference();
+      }
+   TR_ASSERT_FATAL(tempSlotForCallResult, "Symbol reference for store node can't be null\n");
+
+   // Ready to create our fallthrough block now. Connect it to ifCmpTreeTop, and then split it with nodes commoned.
+   int32_t byteArrayType = fej9->getNewArrayTypeFromClass(fej9->getByteArrayClass());
+
+   // Create a new arraylength node.
+   TR::Node *lenNode = TR::Node::create(node, TR::arraylength, 1, sourceArrayNode);
+
+   // Create the destination array
+   TR::Node *destinationArrayNode = TR::Node::createWithSymRef(node, TR::newarray, 2, comp()->getSymRefTab()->findOrCreateNewArraySymbolRef(node->getSymbolReference()->getOwningMethodSymbol(comp())));
+   destinationArrayNode->setAndIncChild(0, lenNode);
+   destinationArrayNode->setAndIncChild(1, TR::Node::iconst(byteArrayType));
+   TR::TreeTop *destinationArrayTreeTop = TR::TreeTop::create(comp(), TR::Node::create(node, TR::treetop, 1, destinationArrayNode));
+   ifCmpTreeTop->insertAfter(destinationArrayTreeTop);
+   destinationArrayNode->setCanSkipZeroInitialization(true);
+   destinationArrayNode->setIsNonNull(true);
+
+   // We have the length node, and also the destination array. Now we create the arraytranslate IL to do the encoding operation.
+   /*
+   // tree looks as follows:
+   // arraytranslate
+   //    input ptr
+   //    output ptr
+   //    translation table
+   //    termCharNode i.e stop character for table based methods
+   //    input length (in elements)
+   //    stoppingNode for SIMD/non-table based methods. Can take on values from -1 to 0xFFFF
+   */
+
+   // Note: The termCharNode and stoppingNode children are ignored for the encodeASCII case.
+
+   TR::Node *arraytranslateNode = TR::Node::create(node, TR::arraytranslate, 6);
+   arraytranslateNode->setSymbolReference(comp()->getSymRefTab()->findOrCreateArrayTranslateSymbol());
+
+   TR::Node *newInputNode = TR::Node::create(sourceArrayNode, comp()->target().is64Bit() ? TR::aladd : TR::aiadd, 2);
+   TR::Node *newInputChild2Node = TR::Node::create(sourceArrayNode, comp()->target().is64Bit() ? TR::lconst : TR::iconst, 0, (int32_t)TR::Compiler->om.contiguousArrayHeaderSizeInBytes());
```

Comment:
Is it possible to use an in64_t for the constant value as is expected usually for an lconst ?

Reply:
Fixed it here: 833e6d726a1